cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -105,22 +105,19 @@ Result<TypeHolder> ResolveGroupOutputType(KernelContext* ctx,
 }
 
 HashAggregateKernel MakeKernel(std::shared_ptr<KernelSignature> signature,
-                               KernelInit init) {
-  HashAggregateKernel kernel;
-  kernel.init = std::move(init);
-  kernel.signature = std::move(signature);
-  kernel.resize = HashAggregateResize;
-  kernel.consume = HashAggregateConsume;
-  kernel.merge = HashAggregateMerge;
-  kernel.finalize = HashAggregateFinalize;
+                               KernelInit init, const bool ordered = false) {
+  HashAggregateKernel kernel(std::move(signature), std::move(init), HashAggregateResize,
```

Comment:
same as original code, but refactored a bit so that we cannot have an partially initialized kernel here (the constructor makes sure of that)